GooglePay/src/main/java/com/braintreepayments/api/GooglePayReadinessResult.kt

```diff
@@ -0,0 +1,19 @@
+package com.braintreepayments.api
+
+/**
+ * Result of [GooglePayClient.isReadyToPay]
+ */
+sealed class GooglePayReadinessResult {
+
+    /**
+     * The Google Pay API is supported and set up on this device. Show the Google Pay button for
+     * Google Pay.
+     */
+    object ReadyToPay : GooglePayReadinessResult()
+
+    /**
+     * The Google Pay API is supported or not set up on this device, or there was an [error]
+     * determining readiness.
+     */
+    class NotReadyToPay(val error: Exception?) : GooglePayReadinessResult()
```

Comment:
Decided to combine not ready with any failures fetching readiness - but can split them out into two types if that's preferred.